api/strapi-api.ts

```
@@ -435,3 +437,82 @@ export const updateStudyStreakStrapi = async (
 
   return { message: "Study streak updated successfully" };
 };
+
+/**
+ * Subscribes a student to a course by creating a course enrollment relation.
+ *
+ * @param userId - The student ID
+ * @param courseId - The course ID
+ * @returns Updated student object
+ */
+export const subscribeCourseStrapi = async (
+  userId: string,
+  courseId: string,
+): Promise<Student> => {
+  // Create a new course enrollment relation
+  const enrollmentDate = new Date().toISOString().split("T")[0]; // Format: YYYY-MM-DD
+
+  const enrollmentResponse =
+    await courseEnrollmentRelationPostCourseEnrollmentRelations({
+      body: {
+        data: {
+          student: userId,
+          course: courseId,
+          enrollmentDate,
+        },
+      },
+    });
+
+  if (!enrollmentResponse?.data) {
+    throw new Error("Failed to subscribe to course");
+  }
```

This could properly be wrapped under a try catch statement, to cover both missing data cases and network cases for better error handling.

---

I believe the current approach is cleaner as we don’t care about wrapping the error message and just want to fail fast so i skip the try/catch entirely: